packages/workflow-upgrade/index.js

```diff
@@ -7,52 +7,78 @@ const rimraf = require('rimraf');
 const Logger = require('@availity/workflow-logger');
 
 const asyncExec = promisify(exec);
-const reinstallTimeout = 30 * 1000 * 60; // 30 minutes
+
+const REINSTALL_TIMEOUT = 30 * 1000 * 60; // 30 minutes
+const MAX_BUFFER_SIZE = 1024 * 5000; // 5MB
+
+const reinstallNodeModules = async (installer, peerInfoReceived) => {
+  Logger.info('Reinstalling packages..');
+
+  // Run install command
+  await asyncExec(`${installer} install`, { timeout: REINSTALL_TIMEOUT, maxBuffer: MAX_BUFFER_SIZE }, () => {
+    Logger.success('\nCongratulations! Welcome to the new @availity/workflow.');
+
+    if (!peerInfoReceived) {
+      Logger.warn(
+        'To complete your upgrade, please install the peerDependencies from eslint-config-availity as devDependencies in your project.'
+      );
+    }
+  });
+};
+
+const getLatestNpmVersion = async (pkgName, defaultVersion) => {
+  try {
+    let { stdout } = await asyncExec(`npm view ${pkgName} version`);
+    return stdout.trim();
+  } catch {
+    Logger.warn(`There was an error getting the latest ${pkgName} version. Defaulting to ${defaultVersion}`);
+    latestWorkflowVersion = defaultVersion;
+  }
+};
 
 module.exports = async (cwd) => {
   Logger.info('Upgrading @availity/workflow');
   const pkgFile = path.join(cwd, 'package.json');
 
   if (fs.existsSync(pkgFile)) {
     // Read Package File to JSON
+    Logger.info('Reading package.json...');
     const pkg = readPkg.sync({ cwd, normalize: false });
     const { devDependencies, scripts, availityWorkflow } = pkg;
+
     const pkgLock = path.join(cwd, 'package-lock.json');
     const yarnLock = path.join(cwd, 'yarn.lock');
+
+    const hasPkgLock = fs.existsSync(pkgLock);
+    const hasYarnLock = fs.existsSync(yarnLock);
+
     let installer = '';
     let peerInfoReceived = false;
 
-    if (fs.existsSync(pkgLock)) {
-      // delete package lock, set npm as installer
-      Logger.info('Deleting Package-Lock');
-      fs.unlinkSync(pkgLock);
+    // Get latest verison of packages
+    let latestWorkflowVersion = await getLatestNpmVersion('@availity/workflow', '10.0.0');
+    let latestEslintVersion = await getLatestNpmVersion('eslint-config-availity', '9.0.0');
+
+    // Determine which pkg manager is used
+    Logger.info('Determining which package manager is being used...');
+    if (hasPkgLock) {
```

Comment:
Do we think npm is still the most commonly used package manager? I've definitely see projects with both. Maybe we should check if both exist and, if so, ask which to use (and recommend deleting the other)

Reply:
I'm down to explore this on it's own though

Reply:
I think that's a good idea. I'll see what I can do

Reply:
I found a package called `inquirer` which made it pretty easy to add